cmake/CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+# The following lines of boilerplate have to be in your project's
```

Comment:
this file can move up to root directory.

Reply:
the cmake build system expects a certain directory structure. If we move this file up the 'main' directory has to move also. Are you ok with that?

Reply:
On let's keep it here for now. There should be a way to have a file at the top level though. That can come later though

Reply:
Where is the requirement for directory structure coming from? It doesn't appear to be cmake itself as the ESP-IDF project has a top level CMakeLists.txt file which is the entry point which then goes into the modules for building etc.

Reply:
Sorry - not requirement, more a directory structure used by default. You can see it here https://github.com/espressif/esp-iot-solution/tree/master/examples. With the VisualGDB plugin I use the directory structure is automatically created and the IDF cmake system adds the 'main' directory (esp-idf/v3.2/tools/cmake/idf_functions.cmake) to the project. I can look into this further and I am sure it is possible to change.